src/main/kotlin/org/jitsi/jibri/api/xmpp/XmppApi.kt

```diff
@@ -0,0 +1,245 @@
+    private val xmppConfigs: List<XmppEnvironmentConfig>
+) {
+    private val logger = Logger.getLogger(this::class.qualifiedName)
+    private val executor = Executors.newSingleThreadExecutor(NameableThreadFactory("XmppApi"))
+
+    /**
+     * Start up the XMPP API by connecting and logging in to all the configured XMPP environments.  For each XMPP
+     * connection, we'll listen for incoming [JibriIq] messages and handle them appropriately.  Join the MUC on
+     * each connection and send an initial [JibriStatusPacketExt] presence.
+     */
+    fun start() {
+        JibriStatusPacketExt.registerExtensionProvider()
+        ProviderManager.addIQProvider(
+            JibriIq.ELEMENT_NAME, JibriIq.NAMESPACE, JibriIqProvider()
+        )
+
+        // Join all the MUCs we've been told to
+        for (config in xmppConfigs) {
+            for (host in config.xmppServerHosts) {
+                logger.info("Connecting to xmpp environment on $host with config $config")
+                val configBuilder = XMPPTCPConnectionConfiguration.builder()
+                    .setHost(host)
+                    .setXmppDomain(config.controlLogin.domain)
+                    .setUsernameAndPassword(config.controlLogin.username, config.controlLogin.password)
+                if (config.trustAllXmppCerts) {
+                    logger.info("The trustAllXmppCerts config is enabled for this domain, " +
+                            "all XMPP server provided certificates will be accepted")
+                    configBuilder.setCustomX509TrustManager(TrustAllX509TrustManager())
+                    configBuilder.setHostnameVerifier(TrustAllHostnameVerifier())
+                }
+                try {
+                    val mucClient = MucClient(configBuilder.build())
+                    mucClient.addIqRequestHandler(object : JibriSyncIqRequestHandler() {
+                        override fun handleJibriIqRequest(jibriIq: JibriIq): IQ {
+                            return handleJibriIq(jibriIq, config, mucClient)
+                        }
+                    })
+                    jibriManager.addStatusHandler { status ->
+                        logger.info("XMPP API got jibri status $status, publishing presence to connection ${config.name}")
+                        val jibriStatus = JibriPresenceHelper.createPresence(
+                            status,
+                            JidCreate.bareFrom("${config.controlMuc.roomName}@${config.controlMuc.domain}")
+                        )
+                        mucClient.sendStanza(jibriStatus)
+                    }
+                    mucClient.createOrJoinMuc(
+                        JidCreate.entityBareFrom("${config.controlMuc.roomName}@${config.controlMuc.domain}"),
+                        Resourcepart.from(config.controlMuc.nickname)
+                    )
+
+                    val jibriStatus = JibriPresenceHelper.createPresence(
+                        JibriStatusPacketExt.Status.IDLE,
```

Comment:
Perhaps import JibriIq.Status to shorten all the references a little bit?

Reply:
i agree they get a bit long, but with all the various statuses i worry it might get more confusing to not have them be fully qualified.